From 486720e0c2418e7e2e0a16221f7c42a308622254 Mon Sep 17 00:00:00 2001
From: Jes Sorensen <jsorensen@fb.com>
Date: Tue, 9 Jul 2019 14:49:22 -0400
Subject: [PATCH] super-intel: Use put_unaligned in split_ull
Git-commit: 486720e0c2418e7e2e0a16221f7c42a308622254
Patch-mainline: mdadm-4.1+
References: jsc#SLE-10078, jsc#SLE-9348

Shut up some gcc9 errors by using put_unaligned() accessors. Not pretty,
but better than it was.

Also correct to the correct swap macros.

Signed-off-by: Jes Sorensen <jsorensen@fb.com>
Signed-off-by: Coly Li <colyli@suse.de>

---
 super-intel.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/super-intel.c b/super-intel.c
index 230e164..d7e8a65 100644
--- a/super-intel.c
+++ b/super-intel.c
@@ -1165,12 +1165,12 @@ static int count_memberships(struct dl *dl, struct intel_super *super)
 
 static __u32 imsm_min_reserved_sectors(struct intel_super *super);
 
-static int split_ull(unsigned long long n, __u32 *lo, __u32 *hi)
+static int split_ull(unsigned long long n, void *lo, void *hi)
 {
 	if (lo == 0 || hi == 0)
 		return 1;
-	*lo = __le32_to_cpu((unsigned)n);
-	*hi = __le32_to_cpu((unsigned)(n >> 32));
+	__put_unaligned32(__cpu_to_le32((__u32)n), lo);
+	__put_unaligned32(__cpu_to_le32((n >> 32)), hi);
 	return 0;
 }
 
-- 
2.25.0

